mtags/src/main/scala-2/scala/meta/internal/pc/ConvertToNamedArgumentsProvider.scala

```diff
@@ -0,0 +1,37 @@
+package scala.meta.internal.pc
+
+import scala.meta.pc.OffsetParams
+
+import org.eclipse.{lsp4j => l}
+
+final class ConvertToNamedArgumentsProvider(
+    val compiler: MetalsGlobal,
+    params: OffsetParams,
+    numUnnamedArgs: Int
+) {
+
+  import compiler._
+  def convertToNamedArguments: List[l.TextEdit] = {
+    val unit = addCompilationUnit(
+      code = params.text(),
+      filename = params.uri().toString(),
+      cursor = None
+    )
+
+    val typedTree = typedTreeAt(unit.position(params.offset))
+    typedTree match {
+      case Apply(fun, args) =>
+        args
+          .take(numUnnamedArgs)
```

Comment:
What if we have a named argument in the middle like

```scala
def foo(x: Int, y: Int, z: Int)
foo(1, y = 2, 3)
```

?

Reply:
I guess we can handle this case even though it's not valid syntax, but it's not clear what the behavior should be if the existing named arg is not in the same order as the parameter declaration e.g.
```scala
def foo(x: Int, y: Int, z: Int)
foo(1, z = 2, 3)
```